platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateMap.java

```diff
@@ -47,41 +52,58 @@ public SignedStateMap() {
      * @return an auto-closable object that wraps a signed state. May point to a null state if there is no state for the
      * given round. Will automatically release the state when closed.
      */
-    public synchronized AutoCloseableWrapper<SignedState> get(final long round) {
-
-        final SignedState signedState = map.get(round);
+    public synchronized @NonNull AutoCloseableWrapper<SignedState> get(final long round) {
+        return newSignedStateWrapper(map.get(round));
+    }
 
-        return newSignedStateWrapper(signedState);
+    /**
+     * Get the latest state in this map. A reservation is taken on the state before this method returns.
+     *
+     * @return an auto-closable object that wraps a signed state. May point to a null state if there are no states in
+     * this container. Will automatically release the state when closed.
+     */
+    public synchronized @NonNull AutoCloseableWrapper<SignedState> getLatest() {
```

Comment:
Can you use a Lock object instead of `synchronized`? That is the preferred way in Java and we even have some additional features by using our AutoClosableLock. Please use Locks class as facade for creation ;)

Reply:
converted class to use explicit lock